### ISSUE TITLE
Rewrite automatic unit insertion

### ISSUE THREAD

**Issue by gabro** [scalacenter/scalafix#214]
```scala
// before
val x: Option[Unit] = Option()

// after
val x: Option[Unit] = Option(())
```

As discussed in #208, we should include this into `NoAutotupling`, and rename it to `NoAdaptedArgs` in order to match the scalac flag that detects both cases.

The implementation should be simple enough, since the compiler provides detailed messages for this case too.

**Comment by olafurpg** [comment]
This would be a nice rewrite. Question is whether we should bundle this rewrite with NoAutoTupling or keep it separate. I think it might make more sense to keep it separate as `NoUnitInsertion`. We can compose the two rewrites 

```scala
val NoAdaptedArgs = NoUnitInsertion.andThen(NoAutoTupling)
```

**Comment by olafurpg** [comment]
For rewrites like this it might be useful to have some kind of `PositionPatch` that works on positions, that would be an even lower-level patch than token patches. What do you think?

**Comment by gabro** [comment]
> We can compose the two rewrites
> 
> ```scala
> val NoAdaptedArgs = NoUnitInsertion.andThen(NoAutoTupling)
> ```

I like that! Composition FTW 👍 

> For rewrites like this it might be useful to have some kind of PositionPatch that works on positions

Yes, especially for rewrites working with compiler messages, some position-based operations would be extremely useful!

**Comment by gabro** [comment]
I also think this would be a good first contribution, taking `NoAutotupling` as reference implementation.

**Comment by olafurpg** [comment]
Fixed in #352
